src/game/OperationHandling.hpp

```diff
@@ -120,51 +112,92 @@ namespace actions {
-                fsm.remainingCharacters.pop_front();
-            }
         }
     };
 
     /**
-     * Request operation from next character in list, emits events::triggerNPCmove if next is NPC
+     * Chooses next Character and requests Operation.
+     * Emits events::triggerNPCmove, triggerCatMove, triggerJanitorMove, roundDone
      */
     struct requestNextOperation {
         template<typename Event, typename FSM, typename SourceState, typename TargetState>
-        void operator()(Event &&, FSM &fsm, SourceState &, TargetState &) {
-            spdlog::trace("requestNextOperation for character {}", fsm.activeCharacter);
+        void operator()(Event &&event, FSM &fsm, SourceState &, TargetState &) {
+            spdlog::info("RequestNextOperation: last active character was {}", fsm.activeCharacter);
             spy::gameplay::State &state = root_machine(fsm).gameState;
-            using spy::util::RoundUtils;
 
-            if (RoundUtils::isGameOver(state)) {
-                // There are still characters remaining, but the game has been won with the last action
-                // We do not have to request a new operation, and abort early
+            if (spy::util::RoundUtils::isGameOver(state)) {
+                // There may still be characters remaining, but the game has been won with the last action.
+                // We do not have to request a new operation, and abort early.
                 spdlog::info("Skipping requestNextOperation because game is already over.");
                 return;
             }
 
+            // Check if the current character has retired
+            bool isRetire = false;
+            if constexpr(std::is_same<Event, spy::network::messages::GameOperation>::value) {
+                const spy::network::messages::GameOperation &operation = event;
+                if (operation.getOperation()->getType() == spy::gameplay::OperationEnum::RETIRE) {
+                    isRetire = true;
+                }
+            }
+
+            // Check if the next character has to be chosen or not
+            bool advanceCharacter = true;
+            if (fsm.activeCharacter != spy::util::UUID{}) {
+                // Some character was already active this round
+                if (state.getCharacters().findByUUID(fsm.activeCharacter) != state.getCharacters().end()) {
+                    spdlog::debug("Last character was a regular character that might make another action");
+
+                    const spy::character::Character &activeCharacter = *state.getCharacters().findByUUID(
+                            fsm.activeCharacter);
+
+                    if (not isRetire and Util::canMoveAgain(activeCharacter)) {
+                        spdlog::info("Character {} has not retired and can make another move.",
+                                     activeCharacter.getName());
+                        advanceCharacter = false;
+                    }
```

Comment:
Hier könnte man die AP und BP noch auf Null setzen, dann ist das für die anderen Clients auch im GameStatus transparent, dass der Charakter fertig ist.

Reply:
Das passiert bereits beim Ausführen der Retire action in `ActionExecutor::executeRetire`